eng/scripts/Pack-Crates.ps1

```diff
@@ -158,61 +197,77 @@ function Create-ApiViewFile($package) {
 
 Push-Location $RepoRoot
 try {
-  $localRegistryPath = Initialize-VendorDirectory
-
   [array]$packages = Get-PackagesToBuild
 
-  Write-Host "Building packages in the following order:"
+  $command = "cargo +nightly -Zpackage-workspace package --allow-dirty --locked"
+
+  Write-Host "Building packages:"
   foreach ($package in $packages) {
     $packageName = $package.name
     $type = if ($package.OutputPackage) { "output" } else { "dependency" }
     Write-Host "  $packageName ($type)"
+    $command += " --package $packageName"
   }
 
-  foreach ($package in $packages) {
-    Write-Host ""
-
-    $packageName = $package.name
-    $packageVersion = $package.version
-
-    $command = "cargo publish --locked --dry-run --package $packageName --registry crates-io --config `"source.crates-io.replace-with='local'`" --config `"source.local.directory='$localRegistryPath'`" --allow-dirty"
+  if (!$Verify) {
+    $command += " --no-verify"
+  }
 
-    if ($NoVerify) {
-      $command += " --no-verify"
-    }
+  if ($env:SYSTEM_DEBUG -eq 'true') {
+    Write-Host "##[group] $RepoRoot/Cargo.lock"
+    Get-Content "$RepoRoot/Cargo.lock"
+    Write-Host "##[endgroup]"
+  }
 
-    Invoke-LoggedCommand -Command $command -GroupOutput
+  Invoke-LoggedCommand -Command $command -GroupOutput
 
+  if ($env:SYSTEM_DEBUG -eq 'true') {
+    Write-Host "##[group] $RepoRoot/Cargo.lock"
+    Get-Content "$RepoRoot/Cargo.lock"
+    Write-Host "##[endgroup]"
+  }
```

Comment:
This shouldn't be necessary, but if you were going to compare before and after, why not do that in the script and not make the human save to files just to diff? Back up the `Cargo.lock` from before, do the new, and emit `git diff -- file1 file2` or something.

Reply:
I actually want to see the contents of the lock file without having to download anything, not just the diff